### ISSUE TITLE
Cannot apply index template with ILM enabled

### ISSUE THREAD

**Issue by gboddin** [elastic/beats#12611]
Version : 7.1.1

Config:

```
setup:
  ilm:
    enabled: true
    policy_name: "biglogindex-ilm-policy"
    policy_file: "/conf/ilm.json"
    rollover_alias: "biglogindex"
    pattern: "{now/d}-000001"
    overwrite: true
  template:
    enabled: true
    name: "biglogindex"
    fields: "/conf/template.yml"
    pattern: "biglogindex-*"
    overwrite: true
    settings:
      index:
        number_of_shards: 2
        number_of_replicas: 1
        routing:
          allocation:
            require:
              data: "hot"
```
Fields :

```
mappings:
  properties:
    type:
      type: keyword
    processid:
      type: integer
    statusCode:
      type: integer
```

Expectations : 

- ILM profile is loaded/overwritten
- Index template is loaded/overwitten with ILM policy set
- Logging starts in biglogindex-*

Actual results :
```
2019-06-19T09:48:54.361Z	INFO	[index-management]	idxmgmt/std.go:238	ILM policy successfully loaded.,
2019-06-19T09:48:54.362Z	INFO	[index-management]	idxmgmt/std.go:361	Set setup.template.name to '{biglogindex {now/d}-000001}' as ILM is enabled.,
2019-06-19T09:48:54.362Z	INFO	[index-management]	idxmgmt/std.go:366	Set setup.template.pattern to 'biglogindex-*' as ILM is enabled.,
2019-06-19T09:48:54.362Z	INFO	[index-management]	idxmgmt/std.go:400	Set settings.index.lifecycle.rollover_alias in template to {biglogindex {now/d}-000001} as ILM is enabled.,
2019-06-19T09:48:54.362Z	INFO	[index-management]	idxmgmt/std.go:404	Set settings.index.lifecycle.name in template to {biglogindex-ilm-policy map[policy:map[phases:map[hot:map[actions:map[rollover:map[max_size:50gb max_docs:%!s(float64=3e+06) max_age:1d] set_priority:map[priority:%!s(float64=100)]]] warm:map[min_age:3d actions:map[allocate:map[require:map[data:warm] number_of_replicas:%!s(float64=1)] forcemerge:map[max_num_segments:%!s(float64=4)] shrink:map[number_of_shards:%!s(float64=2)] set_priority:map[priority:%!s(float64=50)]]] cold:map[min_age:30d actions:map[allocate:map[require:map[data:cold] number_of_replicas:%!s(float64=1)] freeze:map[] set_priority:map[priority:%!s(float64=25)]]] delete:map[min_age:365d actions:map[delete:map[]]]]]]} as ILM is enabled.,
2019-06-19T09:48:54.363Z	INFO	template/load.go:82	Loading template for Elasticsearch version: 7.1.1,
2019-06-19T09:48:54.363Z	INFO	template/load.go:84	Existing template will be overwritten, as overwrite is enabled.,
2019-06-19T09:48:54.417Z	INFO	template/load.go:145	Elasticsearch template with name 'biglogindex' loaded,
2019-06-19T09:48:54.417Z	INFO	[index-management]	idxmgmt/std.go:272	Loaded index template.,
2019-06-19T09:48:54.418Z	INFO	[index-management]	idxmgmt/std.go:283	Write alias successfully generated.,
```

Index template is overwritten by ILM, `index.template.fields` is never used, even @timestamp is missing.

Maybe my fields.yml is wrong ? I couldn't find any example of it.

**Comment by gboddin** [comment]
tldr : my file was not in the right format ( you might want to look a the default file in /etc/filebeat/fields.yml on default install ).
Best approach was to use `fields_append` at the moment to avoid duplicating `fields.yml`'s code.
